crates/viewer/re_space_view_spatial/src/ui_3d.rs

```diff
@@ -112,7 +128,28 @@ impl View3DState {
         bounding_boxes: &SceneBoundingBoxes,
         space_cameras: &[SpaceCamera3D],
         scene_view_coordinates: Option<ViewCoordinates>,
+        default_camera: &ViewProperty,
     ) -> ViewEye {
+        if let Ok(Some(pos_origin)) = default_camera.component_or_empty::<CameraOrigin>() {
+            if let Ok(Some(pos_target)) = default_camera.component_or_empty::<CameraTarget>() {
+                if self.view_eye_default.is_none() {
+                    // Facking last eye interaction. Otherwise the next `if`
+                    // statement in this function causes the camera to change
+                    // if there is a change to the bounding box (e.g. when new
+                    // objects are added to the scene.)
+                    self.last_eye_interaction = Some(Instant::now());
+
+                    self.view_eye_default = Some(from_to_eye(
+                        vec3(pos_origin.0),
+                        vec3(pos_target.0),
+                        scene_view_coordinates,
+                    ));
+
+                    self.view_eye = self.view_eye_default;
+                }
+            }
+        }
```

Comment:
I don't quite follow why this is needed.
Shouldn't the behavior of `default_eye` be adjusted instead?

Reply:
When double clicking on the 3D view, I want the camera to jump back to the default camera settings, if specified. The double click calls the `View3DState::reset_camera` method. While the default camera property is available from the gui-rendering path (`default_camera.component_or_empty::<CameraOrigin>()` etc), it is not so easy to query it from the `reset_camera` method. As such, I added a new field `view_eye_default` on the `View3DState`, which is set the first time the gui renders. This field is then accessible from the `View3DState::reset_camera` method.

I know this is not great and it would be cleaner to query to store for the `CameraOrigin` etc component directly, but I was not sure how to do that easily. So I decided to go for this workaround.